mentat/git_handler.py

```diff
@@ -84,17 +85,20 @@ def _get_git_root_for_path(path: Path) -> Path:
         # call realpath to resolve symlinks, so all paths match
         return Path(os.path.realpath(git_root))
     except subprocess.CalledProcessError:
-        logging.error(f"File {path} isn't part of a git project.")
-        raise UserError()
+        if raise_error:
+            logging.error(f"File {path} isn't part of a git project.")
+            raise UserError()
+        else:
+            return
 
 
 def get_shared_git_root_for_paths(paths: list[Path]) -> Path:
     git_roots = set[Path]()
     for path in paths:
-        git_root = _get_git_root_for_path(path)
+        git_root = cast(Path, get_git_root_for_path(path))
```

Comment:
? Why are we casting it here? I assume you're going to change this function in the future (so that len(git_roots) == 0 isn't an error); is this just a placeholder for now? I would rather you change it to only adding the git root into git_roots if it isn't None just to make sure we don't forget in the future (and because this can lead to returning None)